rllab/envs/normalized_gym_env.py

```diff
@@ -0,0 +1,142 @@
+import gym
+import numpy as np
+
+from rllab.core.serializable import Serializable
+from rllab.misc import special
+from rllab.misc.overrides import overrides
+
+
+def gym_space_flatten_dim(space):
+    if isinstance(space, gym.spaces.Box):
+        return np.prod(space.low.shape)
+    elif isinstance(space, gym.spaces.Discrete):
+        return space.n
+    elif isinstance(space, gym.spaces.Tuple):
+        return np.sum([gym_space_flatten_dim(x) for x in space.spaces])
+    else:
+        raise NotImplementedError
+
+
+def flatten_gym_space(obs, space):
+    if isinstance(space, gym.spaces.Box):
+        return np.asarray(obs).flatten()
+    elif isinstance(space, gym.spaces.Discrete):
+        return special.to_onehot(obs, space.n)
+    elif isinstance(space, gym.spaces.Tuple):
+        return np.concatenate(
+            [flatten_gym_space(xi, c) for c, xi in zip(space.spaces, obs)])
+    else:
+        raise NotImplementedError
+
+
```

Comment:
Maybe add flatten_n_gym_space, unflatten_gym_space, and unflatten_n_gym_space?

Reply:
Or we can move these functions into a new helper class. I think that will be beneficial when porting the other rllab.Envs.

Reply:
I agree these would be useful in something like rllab.env.utils